torch/csrc/jit/interpreter.cpp

```diff
@@ -1,42 +1,208 @@
+#include "Python.h"
 #include "interpreter.h"
 #include "torch/csrc/jit/ir.h"
 #include "torch/csrc/autograd/profiler.h"
 #include "torch/csrc/jit/generated/aten_dispatch.h"
+#include "Python.h"
+#include "pybind11/pybind11.h"
+#include "torch/csrc/utils/auto_gil.h"
+#include "torch/csrc/autograd/variable.h"
+#include "torch/csrc/autograd/python_variable.h"
+#include "torch/csrc/autograd/python_engine.h"
+#include "torch/csrc/autograd/functions/special.h"
 #ifdef WITH_CUDA
 #include "torch/csrc/jit/fusion_compiler.h"
 #endif
 
+namespace py = pybind11;
+
 namespace torch { namespace jit {
 
+struct DummyFunction : autograd::Function {
+  DummyFunction() {
+    num_inputs = 0;
+    is_executable = true;
+  }
+  virtual autograd::variable_list apply(const autograd::variable_list& inputs) override {
+    throw std::logic_error("DummyFunction::apply() called, but it should be blocked by a callback returning false");
+  }
+};
+
+struct Handle : at::RefCounted {
+  std::shared_ptr<DummyFunction> forward_inputs;
```

Comment:
Can you add a comment here? I can't see how a "DummyFunction" specifies a list of forward inputs. The way you use it later makes sense, but that's not immediately obvious when you start reading the file.